### PR TITLE
Document some pokenav and region map

### PR DESCRIPTION
**Pull request by GriffinRichards**
NOTE: porymap's region map editor will need to be updated for this

At the moment, these are the only porymap-breaking changes in this PR that I know of
`gPokenavCityMaps[] --> sPokenavCityMaps[NUM_CITY_MAPS]`
`/graphics/pokenav/zoom_tiles.png --> /graphics/pokenav/city_zoom_tiles.png`
pokeruby will need to be updated to fit these as well (and any other unidentified breaking changes)

### CODE REVIEW

**Comment by coltongit** [comment]
Since the Object_Event generalization happened, could we try to document this and get this one out of the way too?

File: src/field_region_map.c [hunk 1]
```diff
@@ -102,8 +112,8 @@ static void MCB2_InitRegionMapRegisters(void)
     ResetSpriteData();
     FreeAllSpritePalettes();
     ResetBgsAndClearDma3BusyFlags(0);
-    InitBgsFromTemplates(1, gUnknown_085E5068, 2);
-    InitWindows(gUnknown_085E5070);
+    InitBgsFromTemplates(1, sFieldRegionMapBgTemplates, 2);
```
Comment (PikalaxALT) [inline src/field_region_map.c]: `2` --> `ARRAY_COUNT(sFieldRegionMapBgTemplates)`

File: src/pokenav.c [hunk 2]
```diff
@@ -51,142 +51,158 @@ static void Task_RunLoopedTask(u8 taskId);
...
-        .unk10 = sub_81CDE64,
-        .unk14 = sub_81CD1C0,
-        .unk18 = sub_81CECA0,
+        .init = PokenavCallback_Init_7,
+        .callback = sub_81CD070,
+        .open = sub_81CDDD4,
+        .createLoopTask = sub_81CDE2C,
+        .isLoopTaskActive = sub_81CDE64,
+        .free1 = sub_81CD1C0,
+        .free2 = sub_81CECA0,
```
Comment (PikalaxALT) [inline src/pokenav.c]: Name these (and below entries)

Comment (GriffinRichards) [reply src/pokenav.c]: They're not related to region map, and I haven't clarified the distinction between the Conditions/Ribbons menus yet (e.g. `POKENAV_MENU_9` or `POKENAV_MENU_A`). I'll do the rest of these when I continue with the rest of pokenav.

**Comment by GriffinRichards** [comment]
Also if it wasn't clear, don't merge this yet. Porymap needs to be updated first

**Pull request merged**